expat/fuzz/xml_lpm_fuzzer.cpp

```diff
@@ -0,0 +1,432 @@
+  XML_StopParser((XML_Parser)userData, XML_TRUE);
+}
+
+static void XMLCALL
+StartCdataSectionHandler(void* userData) {
+}
+
+static void XMLCALL
+EndCdataSectionHandler(void* userData) {
+}
+
+static void XMLCALL
+DefaultHandler(void* userData, const XML_Char* s, int len) {
+  TouchString(s, len);
+}
+
+static void XMLCALL
+StartDoctypeDeclHandler(void* userData, const XML_Char* doctypeName,
+                        const XML_Char* sysid, const XML_Char* pubid,
+                        int has_internal_subset) {
+  TouchString(doctypeName);
+  TouchString(sysid);
+  TouchString(pubid);
+}
+
+static void XMLCALL
+EndDoctypeDeclHandler(void* userData) {
+}
+
+static void XMLCALL
+EntityDeclHandler(void *userData, const XML_Char *entityName,
+                  int is_parameter_entity, const XML_Char *value,
+                  int value_length, const XML_Char *base,
+                  const XML_Char *systemId, const XML_Char *publicId,
+                  const XML_Char *notationName) {
+  TouchString(entityName);
+  TouchString(value, value_length);
+  TouchString(base);
+  TouchString(systemId);
+  TouchString(publicId);
+  TouchString(notationName);
+}
+
+static void XMLCALL
+UnparsedEntityDeclHandler(void *userData, const XML_Char *entityName,
+                          const XML_Char *base, const XML_Char *systemId,
+                          const XML_Char *publicId,
+                          const XML_Char *notationName) {
+  TouchString(entityName);
+  TouchString(base);
+  TouchString(systemId);
+  TouchString(publicId);
+  TouchString(notationName);
+}
+
+static void XMLCALL
+NotationDeclHandler(void *userData, const XML_Char *notationName,
+                    const XML_Char *base, const XML_Char *systemId,
+                    const XML_Char *publicId) {
+  TouchString(notationName);
+  TouchString(base);
+  TouchString(systemId);
+  TouchString(publicId);
+}
+
+static void XMLCALL
+StartNamespaceDeclHandler(void *userData, const XML_Char *prefix,
+                          const XML_Char *uri) {
+  TouchString(prefix);
+  TouchString(uri);
+}
+
+static void XMLCALL
+EndNamespaceDeclHandler(void *userData, const XML_Char *prefix) {
+  TouchString(prefix);
+}
+
+static int XMLCALL
+NotStandaloneHandler(void *userData) {
+  return XML_STATUS_OK;
+}
+
+static int XMLCALL
+ExternalEntityRefHandler(XML_Parser parser, const XML_Char *context,
+                         const XML_Char *base, const XML_Char *systemId,
+                         const XML_Char *publicId) {
+  int rc = XML_STATUS_ERROR;
+  TouchString(context);
+  TouchString(base);
+  TouchString(systemId);
+  TouchString(publicId);
+
+  if (external_entity) {
+    XML_Parser ext_parser = XML_ExternalEntityParserCreate(parser, context,
+                                                           encoding);
+    rc = Parse(ext_parser, (const XML_Char*)external_entity,
+               external_entity_size, 1);
+    XML_ParserFree(ext_parser);
+  }
+
+  return rc;
```

Comment:
It could be beneficial to always return `XML_STATUS_OK` here so that the parser does not abort parsing after this handler.